examples/python/for_satellite_operators/tlm_and_cmd_stream_extended.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2023 Infostellar, Inc.
+# Opens a stream to both receive telemetry and send commands.
+
+from datetime import datetime
+from queue import Queue
+
+from stellarstation.api.v1 import stellarstation_pb2
+from google.protobuf.timestamp_pb2 import Timestamp
+
+import toolkit
+import MY_CONFIG
+
+STREAM_DONE = object()
+
+
+def generate_request(request_queue):
+    # iter(request_queue.get) will block until a message is sent into the queue
+    # we break from the loop by listening for STREAM_DONE to be sent into the queue
+    for request in iter(request_queue.get, None):
+        if request is STREAM_DONE:
+            break
+        yield request
+
+
+def run():
+    # A client is necessary to receive services from StellarStation.
+    client = toolkit.get_grpc_client(
+        MY_CONFIG.API_KEY_PATH, MY_CONFIG.SSL_CA_CERT_PATH)
+
+    # Set up for stream
+    request_queue = Queue()
+    request_generator = generate_request(request_queue)
+    tlm_file = open("tlm_and_cmd_stream_example_tlm.bin", "wb")
+    stream_id = None
+    total_responses = 0
+    total_telemetry_messages = 0
+    total_stream_events = 0
+    total_acks_sent = 0
+    total_messages_sent = 0
+    total_bytes_received = 0
+    command_request_count = 0
+    last_ack_id = None
+
+    # Process responses
+    stop_streaming_critera = [toolkit.PlanLifecycleEventStatus.FAILED]
+    plan_status = toolkit.PlanLifecycleEventStatus.UNKNOWN
+    end_message_received = False
+    stream_attempts = 0
+
+    # Loop lets us try to recover if the connection is unstable
+    while end_message_received is False and \
+            plan_status not in stop_streaming_critera and \
+            stream_attempts < 3:
+        stream_attempts += 1
+
+        # Create init request and add to queue
+        #
+        # stream_id and last_ack_id will be None in the first loop,
+        # but will let us attempt to recover the stream if an error occurs
+        stream_config_request = stellarstation_pb2.SatelliteStreamRequest(
+            satellite_id=str(MY_CONFIG.SATELLITE_ID),
+            enable_events=True,
+            resume_stream_message_ack_id=last_ack_id,
+            stream_id=stream_id,
+            # flow control allows us to resume streams on failures
+            enable_flow_control=True
+            # If you have the plan id, it'll limit your stream to only data for that plan
+            # plan_id=plan_id
+            # if you have the groundstation id, it'll limit your stream to only data from that groundstation
+            # groundstation_id=groundstation_id
+        )
+        request_queue.put(stream_config_request)
+        total_messages_sent += 1
+
+        # Queue a burst of dummy commands (You can send many in a single request - in this case I am sending 10 commands 0xAA)
+        request = stellarstation_pb2.SatelliteStreamRequest(
```

Comment:
The reattempts are a good addition, but that will cause this command burst to be sent on each connection reattempt. I would add logic to make this send just once regardless of number of connection reattempts.